### ISSUE TITLE
plugin doesn't work with RoundCube 1.0

### ISSUE THREAD

**Issue by tofi86** [tofi86/Roundcube-Plugin-Threecol-Layout#5]
@nagyrobi wrote in #4:

I'm also notifying that unfortunately this great plugin doesn't work on Roundcube 1.0. I've tested it in a standalone setup and in a complete kolab environment. See below:
- when the user logs in, the messages list appears shrinked correctly, but the preview panel is missing completely.
- clicking the little arrow on the bottom to close/open it displays a white panel which doesn't contain anything, selecting an e-mail doesn't display it
- clicking again the little arrow on the bottom to close/open the panel will not close it, but will display the message
- from this step on, previewing works, but if the user switches to contacts panel and back to e-mail, he/she has to do all the above again.

Since having three-column display trivial on my setup, unfortunately this prevents me from upgrading to the latest available Roundcube.
Note that on the previous 0.9.5 it works perfectly. It's broken on 1.0.

**Comment by tofi86** [comment]
Neither do I regularly maintain this plugin nor am I the author of it. However I'm also using this plugin frequently and will have a look if I got some free time.
However, this may take some time. Probably someone else is faster and proposing a pull-request?

**Comment by nagyrobi** [comment]
I saw that the original author deleted this plugin from his repository, and I see absolutely no other copy of it elsewhere.

I actually tried to find the problem but failed. That's because although I know some basic php, I don't know anything abouth roundcube's architecture, thus I'm completely lost.
But I feel it shouldn't be too complicated to fix, as it's usable, as long as you click 3 times after the inbox loads.

**Comment by tofi86** [comment]
I see your point, but I'm also not able at the moment to fix this issue... :-/
Probably someone else will send a pull request with a fix...

**Comment by nagyrobi** [comment]
Oh sorry, I really didn't mean to rush you.

**Comment by dpc22** [comment]
Actually the fix appears to be trivial, one of the regexps just needs to be updated:

```
diff --git a/skins/larry/func.php b/skins/larry/func.php
index a4e983d..8283866 100755
--- a/skins/larry/func.php
+++ b/skins/larry/func.php
@@ -9,21 +9,17 @@ function render_page($args)
 {
        $patterns = array();
        $patterns[0] = '/#mailview-top \{ height: [0-9]+px; \}/';
-       $patterns[1] = '/#mailview-bottom \{ top: [0-9]+px; height: auto; \}/';
+       $patterns[1] = '/#mailview-bottom \{ top: [0-9]+px; height: auto; display: block; \}/';
        $patterns[2] = '/id="mailview-top"/';
        $patterns[3] = '/id="mailview-bottom"/';
```

However $patterns[3] and $patterns[4] in the same substitution are now surplus to requirments as Roundcube 1.0.1 no longer has  `<div id="message" class="statusbar">` in the base template.

I can write this up as a pull request if you prefer.

**Comment by aewne** [comment]
Just upgraded to 1.0.1 yesterday. Thanks a bunch for the fix dpc22!

**Comment by tofi86** [comment]
That would be great, @dpc22, as I'm on vacation the next couple of weeks and can't fix it before I leave...

**Comment by dpc22** [comment]
Have done so.

https://github.com/tofi86/Roundcube-Plugin-Threecol-Layout/pull/6

**Comment by tofi86** [comment]
Fixed by #6
